stl/pstl2milp.py

```diff
@@ -0,0 +1,259 @@
+        assert formula.op == Operation.OR
+        z_children = [self.to_milp(f, t, depth+1, z_ancestors) 
+                      for f in formula.children]
+        self.model.addConstr(z == grb.max_(z_children))
+
+    def eventually(self, formula, z, t, depth, z_ancestors):
+        '''Adds an eventually to the model.'''
+        assert formula.op == Operation.EVENT
+        a, b = int(formula.low), int(formula.high)
+        child = formula.child
+        z_children = [self.to_milp(child, t + tau, depth+1, z_ancestors) 
+                      for tau in range(a, b+1)]
+        self.model.addConstr(z == grb.max_(z_children))
+
+    def globally(self, formula, z, t, depth, z_ancestors):
+        '''Adds a globally to the model.'''
+        assert formula.op == Operation.ALWAYS
+        a, b = int(formula.low), int(formula.high)
+        child = formula.child
+        z_children = [self.to_milp(child, t + tau, depth+1, z_ancestors) 
+                      for tau in range(a, b+1)]
+        self.model.addConstr(z == sum(z_children) / (b-a+1))    
+
+    def until(self, formula, z, t, depth, z_ancestors):
+        '''Adds an until to the model.'''
+        a, b = int(formula.low), int(formula.high)
+        z_children = []
+        for t_ in range(a,b+1):
+            z_children_left =  [self.to_milp(formula.left, t+t__, depth+1, 
+                                             z_ancestors) 
+                                for t__ in range(0,t_)]
+            z_children_right = [self.to_milp(formula.right, t+t_, depth+1, 
+                                             z_ancestors)]
+            z_children.append(z_children_right + sum(z_children_left))
+
+        self.model.addConstr(z == grb.max_(z_children))
+
+    def pstl2lp(self, formula, t=0):
+        ''' It creates a linear problem from the formuale
+             that needs to be satisfied '''
+        lp = grb.Model("LP")
+        if self.robust and 'rho' not in self.ranges:
+            self.ranges['rho'] = (-grb.GRB.INFINITY, self.M - 1)
+        if self.robust:
+            rho_min, rho_max = self.ranges['rho']
+            self.rho = lp.addVar(vtype=grb.GRB.CONTINUOUS, name='rho',
+                                        lb=rho_min, ub=rho_max, obj=-1)        
+        else:
+            self.rho = 0
+        
+        formulae = self.predicate_pairs(formula, t)
+        self.lpvariable = dict()
+
+        for phi in formulae:
+            self.lpvariable[phi[0]] = dict()
+            name = '{}_{}'.format(phi[0], phi[1])
+            var = lp.addVar(vtype=grb.GRB.CONTINUOUS, lb=-10, ub=10, name=name)
+            self.lpvariable[phi[0]][phi[1]] = var
+
+            if phi[0].relation in (RelOperation.GE, RelOperation.GT):
+                lp.addConstr(var >= phi[0].threshold + self.rho)
+
+            if phi[0].relation in (RelOperation.LE, RelOperation.LT):
+                lp.addConstr(var <= phi[0].threshold - self.rho)    
+            lp.update()
+        
+        lp.optimize()
```

Comment:
Perhaps make optimization optional using an extra parameter to the method with default value `True` to perform the optimization.